allow extra files to be made available to the executor

eli-jordan opened the pull request:
This change allows extra files to be made available to the executor, by allowing an extraUri parameter to be specified in the query parameters when adding the broker. Any values in the extraUri parameter are then converted into a mesos fetcher URI, so these files are fetched before invoking the executor. 

This allows, for example, to make the jar file for a java agent available when launching the broker.

This addresses https://github.com/mesos/kafka/issues/205

eli-jordan committed:
allow extra files to be made available to the executor

joestein commented:
@eli-jordan this looks like a great feature want to get it into the 0.9.5.1 release do you mind updating readme? please? so folks know how to use it? Throw in a test too so it doesn't break along the way?

eli-jordan committed:
add doc + unit tests + support in build-image.sh for extra files

eli-jordan commented:
@joestein Sure thing, I just wanted to make sure this feature made sense before investing too much time. I have added some doc, added support to the CLI and the build-image.sh file, and also added some tests.

steveniemitz commented:
Whats the status on this?  I had hacked something together to do something similar to this also.
